### PR TITLE
Add web_port to the CR

### PR DESCRIPTION
**Pull request by cfcosta**
Followup to https://github.com/kiali/kiali/pull/3383.

### CODE REVIEW

File: molecule/null-cr-values-test/kiali-cr.yaml [hunk 1]
```diff
@@ -138,6 +138,7 @@ spec:
     metrics_port: null
     port: null
     web_fqdn: null
+    web_port: null
```
Comment (israel-hdez) [inline molecule/null-cr-values-test/kiali-cr.yaml:141]: I think you should move this (and also the other places) to appear after `web_history_mode` to keep alphabetization.
Right, @jmazzitelli ?

**Review by israel-hdez — CHANGES_REQUESTED** [review]

File: molecule/null-cr-values-test/kiali-cr.yaml (hunk 1 above)
Comment (jmazzitelli) [reply molecule/null-cr-values-test/kiali-cr.yaml:141]: correct... a peer settings within the same group should be sorted alphabetically.

Comment (cfcosta) [reply molecule/null-cr-values-test/kiali-cr.yaml:141]: Can do, I kept them together because they are intrinsically related (changing both seems to be the correct default), but alphabetizing might be better.

File: molecule/null-cr-values-test/kiali-cr.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ spec:
   additional_display_details: null
   installation_tag: null
   istio_component_namespaces: null
-  istio_namespace: {{ istio.control_plane_namespace }}
+  istio_namespace: { { istio.control_plane_namespace } }
```
Comment (jmazzitelli) [inline molecule/null-cr-values-test/kiali-cr.yaml:11]: This file got all screwed up. First, this is incorrect Ansible and won't run. No spaces between the two braces are allowed. Looks like this happened all throughout this file.

File: roles/default/kiali-deploy/defaults/main.yml [hunk 3]
```diff
@@ -14,18 +14,18 @@ kiali_defaults:
   version: "default"
 
   additional_display_details:
-  - title: "API Documentation"
-    annotation: "kiali.io/api-spec"
-    icon_annotation: "kiali.io/api-type"
+    - title: "API Documentation"
+      annotation: "kiali.io/api-spec"
+      icon_annotation: "kiali.io/api-type"
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:19]: Something reformated with the wrong indentation. All operator yamls do not "double-indent". All array list items have their "-" marker on the same level as the parent. This bad reformatting is done all throughout this file.

File: roles/default/kiali-deploy/defaults/main.yml [hunk 4]
```diff
@@ -45,15 +45,16 @@ kiali_defaults:
     strategy: ""
 
   deployment:
-    accessible_namespaces: ["^((?!(istio-operator|kube.*|openshift.*|ibm.*|kiali-operator)).)*$"]
+    accessible_namespaces:
+      ["^((?!(istio-operator|kube.*|openshift.*|ibm.*|kiali-operator)).)*$"]
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:49]: revert - looks like an IDE wrapped this line.

File: roles/default/kiali-deploy/defaults/main.yml [hunk 5]
```diff
@@ -45,15 +45,16 @@ kiali_defaults:
...
     #additional_service_yaml:
     affinity:
       node: {}
       pod: {}
       pod_anti: {}
     custom_dashboards:
-      excludes: ['']
-      includes: ['*']
+      excludes: [""]
+      includes: ["*"]
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:57]: technically OK, but why is this changed?

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
looks like an IDE reformatted this all wrong :)

**Comment by cfcosta** [comment]
@jmazzitelli fixed, weird :(

File: roles/default/kiali-deploy/defaults/main.yml [hunk 6]
```diff
@@ -225,6 +225,7 @@ kiali_defaults:
     port: 20001
     web_fqdn: ""
     web_history_mode: ""
+    web_port: 20001
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:228]: Is it true we want to hardcode the default to 20001? I read the description of this value and it says, "When empty, Kiali will try to guess this value from HTTP headers." As I understand, that's the current behavior (attempts to guess the port).

I'm just confirming that we no longer want "guessing" to be the default behavior and that we will now always assume 20001 by default unless the person sets this value to "" in the Kiali CR.

Comment (israel-hdez) [reply roles/default/kiali-deploy/defaults/main.yml:228]: I think we DO want the default behavior of "guessing" the public port.

I re-checked the changes in the back-end and I see that if "guessing" fails, it fallbacks to the listening port in the pod (which is 20001 by default).

So, I don't think we want a fixed value here. It will break existing installs that are relying on Kiali to "guess" the public port.

File: deploy/kiali/kiali_cr.yaml [hunk 7]
```diff
@@ -744,6 +744,14 @@ spec:
 #    ---
 #    web_history_mode: ""
 #
+# Defines the ingress port where the connections come from. This is usually
+# necessary when the application responds through a proxy/ingress, and it does
+# not forward the correct headers so Kiali can guess the port.
+#
+# When empty, Kiali will try to guess this value from HTTP headers.
+#    ---
+#    web_port: 20001
```
Comment (jmazzitelli) [inline deploy/kiali/kiali_cr.yaml:753]:
```suggestion
#    web_port: ""
```

File: roles/default/kiali-deploy/defaults/main.yml (hunk 6 above)
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:228]:
```suggestion
    web_port: ""
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
default to "guess the port" config value

File: roles/default/kiali-deploy/defaults/main.yml (hunk 6 above)
Comment (jmazzitelli) [reply roles/default/kiali-deploy/defaults/main.yml:228]: Note - since this appears to be a numeric value - do we want the default to be `0` or empty string `""`? I don't know what the backend will want.
@israel-hdez @cfcosta

**Comment by cfcosta** [comment]
@jmazzitelli from the code, it's defined as a string, so your suggestions work.

**Review by jmazzitelli — APPROVED** [review]
LGTM

**Comment by israel-hdez** [comment]
I'll try this shortly.

**Review by israel-hdez — APPROVED** [review]
I'm approving these changes, but rejecting the server side PR.
Sorry about that. See: https://github.com/kiali/kiali/pull/3383#discussion_r522408016

**Comment by cfcosta** [comment]
@israel-hdez no problems! Added a fix there, when it is ok we can merge both together.
